src/optimizer/rule/empty_prefix_removal.cpp

```diff
@@ -0,0 +1,47 @@
+#include "duckdb/optimizer/rule/empty_prefix_removal.hpp"
+
+#include "duckdb/execution/expression_executor.hpp"
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/planner/expression/bound_constant_expression.hpp"
+
+#include <regex>
+
+using namespace duckdb;
+using namespace std;
+
+EmptyPrefixRemovalRule::EmptyPrefixRemovalRule(ExpressionRewriter &rewriter) : Rule(rewriter) {
+	// match on a FunctionExpression that has a foldable ConstantExpression
+	auto func = make_unique<FunctionExpressionMatcher>();
+	func->matchers.push_back(make_unique<ExpressionMatcher>());
+	func->matchers.push_back(make_unique<ExpressionMatcher>());
+	func->policy = SetMatcher::Policy::SOME;
+	func->function = make_unique<SpecificFunctionMatcher>("prefix");
```

Comment:
Perhaps also handle SUFFIX/CONTAINS here?

Reply:
that's a great idea. I had no idea these functions existed :)